File: Java/src/com/mergesort/StMergeSort.java
package com.mergesort;

public class StMergeSort {
  public static void mergeSort(Integer[] arr, int start, int end) {
    if (start < end) {
      int mid = start + (end - start) / 2;
      mergeSort(arr, start, mid);
      mergeSort(arr, mid + 1, end);
      MergeUtils.merge(arr, start, mid, end);
    }
  }
}

File: C++/src/st_merge_sort.cpp
#include "st_merge_sort.hpp"
#include "merge_utils.hpp"

void StMergeSort::merge_sort(std::vector<int> &vec, size_t start, size_t end)
{
  if (start < end)
  {
    size_t mid = (start + end) / 2;
    merge_sort(vec, start, mid);
    merge_sort(vec, mid + 1, end);
    MergeUtils::merge(vec, start, mid, end);
  }
}